File: src/UglyNumber.java
public class UglyNumber {
    public static boolean isUgly(int number) {
        if (number <= 0) {
            return false;
        }

        number = divideBy(number, 2);
        number = divideBy(number, 3);
        number = divideBy(number, 5);
        return number == 1;
    }

    private static int divideBy(int number, int primeFactor) {
        while (number % primeFactor == 0) {
            number /= primeFactor;
        }
        return number;
    }
}

File: python/ugly_number.py
class Solution:
    def eliminate_prime_factor(self, number: int, prime_factor: int) -> int:
        while number % prime_factor == 0 and number != 0:
            number //= prime_factor
        return number

    def isUgly(self, n: int) -> bool:
        n = self.eliminate_prime_factor(n, 2)
        n = self.eliminate_prime_factor(n, 3)
        n = self.eliminate_prime_factor(n, 5)
        return n == 1
